plugins/colorbutton/plugin.js

```diff
@@ -95,6 +95,291 @@ CKEDITOR.plugins.add( 'colorbutton', {
 			} );
 		}
 
+		var ColorBox = CKEDITOR.tools.createClass( {
+			$: function( color, clickFn, label ) {
+				this.$ = new CKEDITOR.dom.element( 'td' );
+				this.color = color;
+				this.label = label || ColorBox.colorNames[ this.color ] || this.color;
+				this.clickFn = clickFn;
+
+				this.setHtml();
+			},
+
+			statics: {
+				colorNames: editor.lang.colorbutton.colors
+			},
+
+			proto: {
+				getElement: function() {
+					return this.$;
+				},
+
+				getHtml: function() {
+					return this.getElement().getOuterHtml();
+				},
+
+				setHtml: function() {
+					this.getElement().setHtml( '<a class="cke_colorbox" _cke_focus=1 hidefocus=true' +
+							' title="' + this.label + '"' +
+							' draggable="false"' +
+							' ondragstart="return false;"' + // Draggable attribute is buggy on Firefox.
+							' onclick="CKEDITOR.tools.callFunction(' + this.clickFn + ',\'' + this.color + '\');' +
+							' return false;"' +
+							' href="javascript:void(\'' + this.color + '\')"' +
+							' data-value="' + this.color + '"' +
+							' role="option">' +
+							'<span class="cke_colorbox" style="background-color:#' + this.color + '"></span>' +
+						'</a>' );
+				},
+
+				setPositionIndex: function( posinset, setsize ) {
+					this.getElement().getChild( 0 ).setAttributes( {
+						'aria-posinset': posinset,
+						'aria-setsize': setsize
+					} );
+				}
+			}
+		} );
+
+		var ColorHistoryRow = CKEDITOR.tools.createClass( {
+			$: function() {
+				this.$ = new CKEDITOR.dom.element( 'tr' );
+				this.$.addClass( 'cke_colorhistory_row' );
+				this.boxes = [];
+			},
+
+			statics: {
+				maxLength: editor.config.colorButton_colorsPerRow
+			},
+
+			proto: {
+				getElement: function() {
+					return this.$;
+				},
+
+				removeLastColor: function() {
+					this.getElement().getLast().remove();
+					return this.boxes.pop();
+				},
+
+				addNewColor: function( colorBox ) {
+					this.boxes.unshift( colorBox );
+					this.getElement().append( colorBox.getElement(), true );
+				},
+
+				extractColorBox: function( colorCode ) {
+					var index = CKEDITOR.tools.getIndex( this.boxes, function( box ) {
+						return box.color === colorCode;
+					} );
+
+					if ( index < 0 ) {
+						return;
```

Comment:
It still returns void, see https://github.com/ckeditor/ckeditor4/pull/3784#discussion_r395652055 (should be changed to `null`).